### PR TITLE
Fixes for federated tests

### PR DESCRIPTION
**Pull request by petervdonovan**
This is an effort to partially correct some of the Python federated test failures that we are observing.

### CODE REVIEW

**Comment by petervdonovan** [comment]
I think that the zombie RTI problem is probably fixed now. I didn't even have to do anything terribly hacky.

Looking at the logs that we are now getting in CI, it seems possible that many (most/all???) of the errors that we are getting could be trivial to fix.

**Comment by petervdonovan** [comment]
The Python federated tests all pass for me locally. CI is failing due to the expired Klighd certificate.

File: org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java [hunk 1]
```diff
@@ -57,11 +58,20 @@ private void mergeImportedConfig(
...
-            this, convertToEmptyListIfNull(targetProperties.getPairs()), errorReporter);
+            this,
+            convertToEmptyListIfNull(targetProperties.getPairs()),
+            getRelativePath(mainResource, federateResource),
+            errorReporter);
       }
     }
   }
 
+  private Path getRelativePath(Resource source, Resource target) {
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java:69]: Isn't there a method in `FedFileConfig` that does something like this?

**Review by lhstrh — APPROVED** [review]
This generally looks good to me! If the tests pass locally, then let's go ahead and merge this into @byeong-gil's branch. Thanks a bunch for helping out with this, @petervdonovan!

File: org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java (hunk 1 above)
Comment (petervdonovan) [reply org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java:69]: Yes, it is similar, and it is better in that it handles important cases that I did not address here. However, it produces paths that are relative to the generated LF file, whereas here we need paths relative to the original main LF file.

The "right way" to fix this is to require the projects consisting of multiple files to specify target properties in a separate Lingo config file. I would be OK with merging this as a stopgap measure until we can do it the right way.

Comment (lhstrh) [reply org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java:69]: Sounds fair to me...

**Review by byeong-gil — APPROVED** [review]
LGTM! Thank you for helping to address these issues, @petervdonovan! I'll merge this to my PR ([lingua-franca#1752](https://github.com/lf-lang/lingua-franca/pull/1752)).

**Pull request merged**